src/waypoint.jsx

```diff
@@ -333,13 +352,18 @@ export default class Waypoint extends React.Component {
    * @return {Object}
    */
   render() {
+    if (this.props.children) {
+      return <div ref={this.refElement}>{this.props.children}</div>;
```

Comment:
Let's just get this "right" from the get-go; `<div>` by default, with an "advanced escape hatch" of `refToMeasure` requiring `React.Children.only` and documentation reminding that a `ref` can't attach to a stateless component. 

Makes usage easy for most people, and accommodates edge cases that are bound to occur. No reason to worry about adding/removing flexibility later.